### PR TITLE
:lipstick: (llm) update old reborn banners

### PR DESCRIPTION
**Pull request by LucasWerey**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - All reborn banners

### 📝 Description

Update old banners to fit with new designs.  + Migrate to newArch
Figma and dev banners are not exactly the same due to react native limitation (shadow on the gradient around the image/ the border) and I am not sure time to dev worths such small improvements

#### Figma :  

| 1        | 2         | 3
| ------------- | ------------- |  ------------- | 
|image|image|image|

#### Result 
- Dark :  

| 1        | 2         | 3
| ------------- | ------------- |  ------------- | 
|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 29 29|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 29 31|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 30 11|
|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 29 21|image|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 30 46|

image

- Light :  

| 1        | 2         | 3
| ------------- | ------------- |  ------------- | 
|image|image|image|

- Other locations : 

| 1        | 2         | 3
| ------------- | ------------- |  ------------- | 
|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 30 51|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 30 57|Simulator Screenshot - iPhone 15 - 2024-11-22 at 10 29 46|

- Smaller device (iphone SE):

| 1        | 2         | 3
| ------------- | ------------- |  ------------- | 
|image|image|image|
### ❓ Context

- **JIRA or GitHub link**: [LIVE-14943](https://ledgerhq.atlassian.net/browse/LIVE-14943)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-14943]: https://ledgerhq.atlassian.net/browse/LIVE-14943?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-mobile/src/components/GradientContainer.tsx [hunk 1]
```diff
@@ -20,19 +21,43 @@ export default function GradientContainer({
...
+    case "bottom-to-top":
+      y1 = "100%";
+      y2 = "0%";
+      break;
+    case "right-to-left":
+      x1 = "100%";
+      x2 = "0%";
+      break;
+  }
+
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/components/GradientContainer.tsx:47]: Can you extract this logic outside of the component, please?
Maybe we can have something like `const positions = getPositions(direction)`
Then we can even spread the result in props:
```
<LinearGradient
            {...positions}
            id="myGradient"
            gradientUnits="userSpaceOnUse"
          >
```
That way we don't redefine the logic at each rerender + it improves readability of the component.

File: apps/ledger-live-mobile/src/components/TabBar/TransferDrawer.tsx [hunk 2]
```diff
@@ -239,39 +239,41 @@ export default function TransferDrawer({ onClose }: Omit<ModalProps, "isRequesti
...
             </Text>
           }
-          style={{ marginTop: 36, paddingTop: 13.5, paddingBottom: 13.5 }}
           buttonLabel={t("buyDevice.bannerButtonTitle2")}
           buttonSize="small"
           event="button_clicked"
           eventProperties={bannerEventProperties}
           screen={screen}
-          {...IMAGE_PROPS_SMALL_NANO}
+          {...IMAGE_PROPS_SMALL_NANO_BOX}
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/components/TabBar/TransferDrawer.tsx:269]: Please always spread objects above other props unless we want to override some existing properties. 
When you spread an object first you make it clear to everyone that we will not override anything from the props declared above.
The only exception is when you explicitly want to override some props.

File: apps/ledger-live-mobile/src/components/GradientContainer.tsx (hunk 1 above)
Comment (LucasWerey) [reply apps/ledger-live-mobile/src/components/GradientContainer.tsx:38]: Done ✅ 
LTMK if it's what u meant

File: apps/ledger-live-mobile/src/components/GradientContainer.tsx [hunk 3]
```diff
@@ -1,40 +1,71 @@
...
   containerStyle,
   gradientStyle,
   children,
   startOpacity = 1,
   endOpacity = 0,
+  direction = "top-to-bottom",
 }: Props) {
   const { colors } = useTheme();
 
+  const positions = useCallback(() => getPositions(direction), [direction]);
```
Comment (KVNLS) [inline apps/ledger-live-mobile/src/components/GradientContainer.tsx:58]: We don't need the `useCallback` here. I think that we can directly do the following:
`const gradientPositions = getPositions(direction);`
If you really have a re-rendering issue, favor useMemo to keep the same object.
Here we keep in memory the function but we always have a new object for `gradientPositions`, so we don't benefit from memorization + as we spread the object in the `LinearGradient` React component, React should be smart enough to compare the string for each value and do not perform the rerender.

**Review by KVNLS — CHANGES_REQUESTED** [review]

**Review by KVNLS — APPROVED** [review]
LGTM!

**Pull request merged**